custom_components/adaptive_lighting/switch.py

```diff
@@ -702,15 +752,41 @@ def __init__(
         self._name = data[CONF_NAME]
         self._lights = data[CONF_LIGHTS]
 
+        for light in data[CONF_LIGHTS]:
+            if not hasattr(self.turn_on_off_listener, "_switches"):
+                self.turn_on_off_listener._switches = {}
+            self.turn_on_off_listener._switches[
+                light
+            ] = self  # light should never be in multiple switches anyway.
```

Comment:
> light should never be in multiple switches anyway

That is actually what I have been doing in my (old / prev house) config. Here I switched the switches on and off such that there was only one switch controlling the light.